public/components/event_analytics/explorer/events_views/data_grid.tsx

```diff
@@ -3,148 +3,271 @@
+
+    const start = datemath.parse(startTime)?.utc().format(DATE_PICKER_FORMAT);
+    const end = datemath.parse(endTime, { roundUp: true })?.utc().format(DATE_PICKER_FORMAT);
+    const tokens = rawQuery.replaceAll(PPL_NEWLINE_REGEX, '').match(PPL_INDEX_INSERT_POINT_REGEX);
+
+    finalQuery = `${tokens![1]}=${
+      tokens![2]
+    } | where ${timeStampField} >= '${start}' and ${timeStampField} <= '${end}'`;
+
+    finalQuery += tokens![3];
+
+    for (let i = 0; i < sortingFields.current.length; i++) {
+      const field = sortingFields.current[i];
+      const dir = field.direction === 'asc' ? '+' : '-';
+      finalQuery = finalQuery + ` | sort ${dir} ${field.id}`;
+    }
+
+    finalQuery =
+      finalQuery +
+      ` | head ${pageFields.current[1]} from ${pageFields.current[0] * pageFields.current[1]}`;
+    getEvents(finalQuery);
+  };
+
+  // setSort and setPage are used to change the query and send a direct request to get data
+  const setSort = (sort: GridSortingColumn[]) => {
+    sortingFields.current = sort;
+    redoQuery();
+  };
+
+  const setPage = (page: number[]) => {
+    pageFields.current = page;
+    redoQuery();
+  };
+
+  // creates the header for each column listing what that column is
+  const dataGridColumns = useMemo(() => {
+    if (storedSelectedColumns.length > 0) {
+      const columns: EuiDataGridColumn[] = [];
+      storedSelectedColumns.map(({ name, type }) => {
+        if (name === 'timestamp') {
+          columns.push(DEFAULT_TIMESTAMP_COLUMN);
+        } else if (name === '_source') {
+          columns.push(DEFAULT_SOURCE_COLUMN);
+        } else {
+          columns.push({
+            id: name,
+            display: name,
+            isSortable: true, // TODO: add functionality here based on type
+          });
+        }
+      });
+      return columns;
+    }
+    return [];
+  }, [storedSelectedColumns]);
+
+  // used for which columns are visible and their order
+  const dataGridColumnVisibility = useMemo(() => {
+    if (storedSelectedColumns.length > 0) {
+      const columns: string[] = [];
+      storedSelectedColumns.map(({ name }) => {
+        columns.push(name);
+      });
+      return {
+        visibleColumns: columns,
+        setVisibleColumns: (visibleColumns: string[]) => {
+          // TODO: implement with sidebar field order (dragability) changes
+        },
+      };
+    }
+    // default shown fields
+    return {
+      visibleColumns: [],
+      setVisibleColumns: () => {},
+    };
+  }, [storedSelectedColumns]);
+
+  // sets the very first column, which is the button used for the flyout of each row
+  const dataGridLeadingColumns = useMemo(() => {
+    return [
+      {
+        id: 'inspectCollapseColumn',
+        headerCellRender: () => null,
+        rowCellRender: ({ rowIndex }: { rowIndex: number }) => {
+          return (
+            <FlyoutButton
+              ref={null}
+              http={http}
+              key={null}
+              docId={'undefined'}
```

Comment:
why defining ref and key to be null? and why is docId `'undefined'`?

Reply:
ref is a vestigial prop that will need to be removed later. 
key and docId are important, but for 'surrounding events', which as a whole isn't implemented yet but will be put in for p1. it won't have an effect on any other part of the flyout currently.